src/uu/pgrep/src/pgrep.rs

```diff
@@ -0,0 +1,300 @@
+// This file is part of the uutils procps package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+pub mod pid;
+
+use clap::{arg, crate_version, Arg, ArgAction, ArgGroup, ArgMatches, Command};
+use once_cell::sync::OnceCell;
+use pid::{walk_pid, PidEntry, TerminalType};
+use regex::Regex;
+use std::{borrow::BorrowMut, cmp::Ordering, collections::HashSet};
+use uucore::{
+    error::{UResult, USimpleError},
+    format_usage, help_about, help_usage,
+};
+
+const ABOUT: &str = help_about!("pgrep.md");
+const USAGE: &str = help_usage!("pgrep.md");
+
+#[uucore::main]
+pub fn uumain(args: impl uucore::Args) -> UResult<()> {
+    let matches = uu_app().try_get_matches_from(args)?;
+    let pattern = collect_arg_patterns(&matches);
+
+    // Start Pattern Check //
+    let flag_newest = matches.get_flag("newest");
+    let flag_oldest = matches.get_flag("oldest");
+
+    if (flag_newest == false && flag_oldest == false) && pattern.is_empty() {
+        return Err(USimpleError::new(
+            2,
+            "no matching criteria specified\nTry `pgrep --help' for more information.",
+        ));
+    }
+
+    if pattern.len() > 1 {
+        return Err(USimpleError::new(
+            2,
+            "only one pattern can be provided\nTry `pgrep --help' for more information.",
+        ));
+    }
+    // End Pattern Check //
+
+    let Some(result) = handle_oldest_newest(&matches).or(handle_matching_pids(&matches)) else {
+        return Err(USimpleError::new(
+            2,
+            "no matching criteria specified\nTry `pgrep --help' for more information",
+        ));
+    };
+
```

Comment:
Referenced from pgrep's manpage, there are might be no processes matched

> EXIT STATUS
> 
> 0
>     One or more processes matched the criteria. For pkill and pidwait, one or more processes must also have been successfully signalled or waited for.
> 1
>     No processes matched or none of them could be signalled.
> 2
>     Syntax error in the command line.
> 3
>     Fatal error: out of memory etc.

How should I do this so that the program exit status is 1?

Reply:
I'm not sure I understand your question correctly: there is a `set_exit_code` function in `uucore`.

Reply:
Yes that't the things I need, thanks :)